### PR TITLE
Make vagrant in CI available

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:
The vagrant package does not seem to be available any more, so we use them from HashiCorp.
#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Ready for review. PTAL @cri-o/cri-o-maintainers

File: scripts/github-actions-setup [hunk 1]
```diff
@@ -1,10 +1,8 @@
 #!/usr/bin/env bash
 set -euox pipefail
 
-apt-get update
-apt-get install -y \
-    libvirt-daemon-system \
-    qemu-kvm \
-    vagrant
+curl -fsSL https://apt.releases.hashicorp.com/gpg | gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
+echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/hashicorp.list
```
Comment (kwilczynski) [inline scripts/github-actions-setup:5]: A small nit. On Debian, the `/etc/apt/keyrings` directory would be for user-managed files.

This is not a blocker.

**Pull request merged**